persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -0,0 +1,154 @@
+            }
+        } else if (rootOfTree instanceof FileObject) {
+            FileObject file = (FileObject) rootOfTree;
+            List<BinaryObject> binaries = file.getBinaryObjects();
+            for (BinaryObject binary : binaries) {
+                destroyTree(binary);
+            }
+        }
+        // destroy root of tree
+        Model stoneModel = rootOfTree.getModel();
+        stoneModel = convertModelToTombstone(rootOfTree);
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+
+        //add premis event to tombstone
+        rootOfTree.getPremisLog().buildEvent(Premis.Deletion)
+        .addImplementorAgent(agent.getUsernameUri())
+        .addEventDetail("Item deleted from repository and replaced by tombstone")
+        .write();
+
+        // persist model
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+    }
+
+    private Model convertModelToTombstone(RepositoryObject destroyedObj) {
+        Model model = destroyedObj.getModel();
+        Resource resc = destroyedObj.getResource();
+
+        if (destroyedObj instanceof BinaryObject) {
+            // whitelisted properties for tombstone model of binaries
+            Selector selectEbucoreFilename = new SimpleSelector(resc, Ebucore.filename, (Object) null);
+            Selector selectDcFilename = new SimpleSelector(resc, DcElements.title, (Object) null);
+            Selector selectPremisOriginalName = new SimpleSelector(resc, Premis.hasOriginalName, (Object) null);
+            Selector selectMimetype = new SimpleSelector(resc, Ebucore.hasMimeType, (Object) null);
+            Selector selectChecksum= new SimpleSelector(resc, Premis.hasMessageDigest, (Object) null);
+            Selector selectFilesize = new SimpleSelector(resc, Premis.hasSize, (Long) null);
+
+            StmtIterator iter = model.listStatements();
+            while (iter.hasNext()) {
+                Statement s = iter.nextStatement();
+                if (!(selectEbucoreFilename.test(s) || selectDcFilename.test(s) || selectPremisOriginalName.test(s)
+                        || selectMimetype.test(s) || selectChecksum.test(s) || selectFilesize.test(s))) {
+                    model.remove(s);
+                }
+            }
+        } else {
+            // retain only the object-type of a non-binary object
+            Selector selectRdfType = new SimpleSelector(resc, RDF.type, (Object) null);
+            StmtIterator iter = model.listStatements();
+            while (iter.hasNext()) {
+                Statement s = iter.nextStatement();
+                if (!selectRdfType.test(s)) {
+                    model.remove(s);
+                }
+            }
+        }
+        // determine path and store in tombstone model
+        String path = pathFactory.getPath(destroyedObj.getPid()).toNamePath();
+        model.add(resc, Cdr.historicalPath, path);
+        model.add(resc, RDF.type, Cdr.Tombstone);
+        return model;
```

Comment:
My impression is that you could populate a new model from the filtered properties rather than editing the existing model and persist the changes back to fedora with no issues with the `repoObjFactory.createObject` approach you're using. I don't feel too strongly about it, seems a little cleaner to not edit the retrieved model though I guess?

Reply:
sure, i could do that. i wrestled with which way made more sense, and for some reason it made more logical sense to me to transform the existing model, since that's the approach we're taking with the object. but i also don't have strong feelings here